internal/lua_tests/lua_api_test.go

```diff
@@ -91,14 +96,19 @@ func TestLuaApi(t *testing.T) {
 		}
 
 		if err := ruleVRCT.DeleteRuntimeTemp(); err != nil {
-			t.Fatal("Failed to remove temporary VRCT files", err.Error())
+			logAndFail(t, "Failed to remove temporary VRCT files: %s", err.Error())
 		}
 
 		if script.afterTest != nil {
 			err = script.afterTest()
 			if err != nil {
-				t.Fatalf("error occured during finalization stage of test '%s': %s", script.file, err)
+				logAndFail(t, "error occured during finalization stage of test '%s': %s", script.file, err)
 			}
 		}
 	}
 }
+
+func logAndFail(t *testing.T, format string, args ...interface{}) {
```

Comment:
What this function is supposed to be? Why not `t.Fatalf`?

Reply:
`t.Fatal` immediately ends execution, like `os.Exit(1)` 

logAndFail, logs and marks test as failed, but without ending execution

It allows test to make a cleanup